crates/evm/src/lib.rs

```diff
@@ -117,6 +120,14 @@ pub trait ConfigureEvmEnv: Send + Sync + Unpin + Clone + 'static {
     /// Fill transaction environment from a [`TransactionSigned`] and the given sender address.
     fn fill_tx_env(&self, tx_env: &mut TxEnv, transaction: &TransactionSigned, sender: Address);
 
+    /// Fill transaction environment with a system contract call.
+    fn fill_tx_env_with_system_contract_call(
```

Comment:
maybe drop the `with` here?
```suggestion
    fn fill_tx_env_system_contract_call(
```